docs/develop/reliable-transaction-submission.md

```diff
@@ -0,0 +1,56 @@
+---
+sidebar_position: 4
+---
+
+# Reliable Transaction Submission
+
+When making payments or recording transfers, it is important to ensure that they are recorded once
+and only once -- even if some parts of the system fail during the transaction.
+
+There are some subtle gotchas to avoid, so this page describes how to submit events -- and
+especially transfers -- reliably.
+
+## The App or Browser Should Generate the ID
+
+[`Transfer`s](../reference/transfers.md#id) and [`Account`s](../reference/accounts.md#id) carry an
+`id` field that is used as an idempotency key to ensure the same object is not created twice.
+
+**The client software, such as your app or web page, that the user interacts with should generate the
+`id` (not your API):**
+
+1. User initiates a transfer.
+2. Client software (app, web page, etc) [generates the transfer `id`](./data-modeling.md#id).
+3. Client software **persists the `id` in the app or browser local storage.**
```

Comment:
We should also describe what to do with that persisted id (since just persisting it does nothing if we don't use it for anything if the page reloads).

Reply:
Good point! I added some more language about this. Lmk what you think